### PR TITLE
Added DB for ride requests

### PR DESCRIPTION
**Pull request by ivanke1**
Added repository and entity classes, removed redundant fullName field

### CODE REVIEW

**Comment by ivanke1** [comment]
foreign key constraints WIP

File: src/main/java/edu/ucsb/cs156/gauchoride/config/SecurityConfig.java [hunk 1]
```diff
@@ -88,6 +88,13 @@ private GrantedAuthoritiesMapper userAuthoritiesMapper() {
           if (email.endsWith("@ucsb.edu")) {
             mappedAuthorities.add(new SimpleGrantedAuthority("ROLE_MEMBER"));
           }
+          
+          Optional<User> u = userRepository.findByEmail(email);
+          if (u.isPresent() && u.get().getDriver()) {
+            mappedAuthorities.add(new SimpleGrantedAuthority("ROLE_DRIVER"));
+          } else {
+            mappedAuthorities.add(new SimpleGrantedAuthority("ROLE_RIDER"));
+          }
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/gauchoride/config/SecurityConfig.java]: Doesn't belong in this PR... 

The database and repository don't depend on it, and that code is already in a separate PR.

Try to develop the habits to keep separate PRs separate.

File: src/main/java/edu/ucsb/cs156/gauchoride/entities/RideRequest.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@Entity(name = "riderequests")
+public class RideRequest {
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private long id;
+  private Long userid;
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/gauchoride/entities/RideRequest.java]: Consider making this riderUserId so that we can distinguish it from driverUserId if we add that later.

File: src/main/java/edu/ucsb/cs156/gauchoride/entities/RideRequest.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  private String day;
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/gauchoride/entities/RideRequest.java]: Please add a comment indicating whether this is MTWRFSU, Mon/Tue/Wed/Thu/Fri/Sat/Sun or fully spelled out days of week (and specify the capitalization if you choose that.)

File: src/main/java/edu/ucsb/cs156/gauchoride/entities/RideRequest.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  private String course;
+  private String startTime;
+  private String stopTime;
+  private String building;
+  private String room;
+  private String pickup;
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/gauchoride/entities/RideRequest.java]: pickupLocation?

Please try to choose field names that remove ambiguity.   String pickup could be lots of things (pickupDriverName, pickupTime, etc.)

File: src/main/java/edu/ucsb/cs156/gauchoride/repositories/RideRequestRepository.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+import edu.ucsb.cs156.gauchoride.entities.RideRequest;
+
+import java.util.Optional;
+
+@Repository
+public interface RideRequestRepository extends CrudRepository<RideRequest, Long> {
+    Iterable<RideRequest> findAllByUserid(Long userid);
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/gauchoride/repositories/RideRequestRepository.java]: If you change this to riderUserId (which I suggest that you do), you'll need to change this too.

**Review by pconrad — DISMISSED** [review]

File: src/main/java/edu/ucsb/cs156/gauchoride/config/SecurityConfig.java (hunk 1 above)
Comment (ivanke1) [reply src/main/java/edu/ucsb/cs156/gauchoride/config/SecurityConfig.java]: Ok. If I'm working on CRUD requests, and was awaiting approval of the PR which was adding driver and rider roles, in that case would it be fine to branch off of the branch awaiting review since they depend on each other

**Review by seannoh — APPROVED** [review]
LGTM

**Review by andrewpengucsb — APPROVED** [review]
LGTM
